synapse/dendrite/jobs.py

```diff
@@ -0,0 +1,214 @@
+import json
+import time
+import logging
+import collections
+import synapse.common as s_common
+import synapse.cortex as s_cortex
+import synapse.eventbus as s_eventbus
+import synapse.telepath as s_telepath
+
+logger = logging.getLogger(__name__)
+
+class Jobs(s_eventbus.EventBus):
+    '''
+    Jobs provides a simple remote queue for managing Parser jobs in Dendrite.
+
+    Parser jobs are added to named queues, an event is fired on the Service Bus, the Parsers registered for
+    the named queue receive the event and proceed to retrieve the job for processing. Basic job lifecycle management
+    is provided where a job can be in one of the following states: queued, working, and completed.
+
+    Args:
+        core (str):
+            URL of the Cortex to be used for internal management of the job queues.
+
+        svcbus (str):
+            URL of the SvcBus that events will be emitted on.
+    '''
+    def __init__(self, core, svcbus):
+        s_eventbus.EventBus.__init__(self)
+        self.currMs = None
+        self.jobNum = None
+        self.core = s_cortex.openurl(core)
+        self.core.addTufoForm('dendrite:job', ptype='guid')
+        self.core.addTufoProp('dendrite:job', 'queue', ptype='str', req=True)
+        self.core.addTufoProp('dendrite:job', 'data', ptype='str', req=True)
+        self.core.addTufoProp('dendrite:job', 'runkey', ptype='int', req=True)
+        self.core.addTufoProp('dendrite:job', 'status', ptype='str', req=True)
+        self.core.addTufoProp('dendrite:job', 'failed_at', ptype='int', req=True)
```

Comment:
cool, I looked briefly at the time type and didn't see an advantage of using it... pleading ignorant here. It seemed to only provide parsing from string, min/max mgmt, and validation. Are you recommending mostly for clearly communicating type? Or, are there other benefits of doing so. Either way, I've made the change... but am curious going forward.